### PR TITLE
Move importer to its own app

### PR DESCRIPTION
**Pull request by dschmidt**
## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-app-importer/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  "description": "ownCloud Web Importer",
+  "license": "AGPL-3.0",
+  "dependencies": {
+    "@uppy/dashboard": "3.4.0",
+    "@uppy/onedrive": "^3.1.1"
+  },
+  "peerDependencies": {
+    "@ownclouders/web-pkg": "workspace:*",
+    "@ownclouders/web-client": "workspace:*",
+    "web-app-files": "workspace:*",
```
Comment (dschmidt) [inline packages/web-app-importer/package.json:14]: TODO: remove

File: packages/web-app-importer/src/extensions.ts [hunk 2]
```diff
@@ -97,33 +87,44 @@ export const useFileActionsImport = ({
...
+      companionUrl
+    })
+    uppyService.addPlugin(GoogleDrive, {
+      target: Dashboard,
+      companionUrl
+    })
+    uppyService.addPlugin(WebdavPublicLink, {
+      target: Dashboard,
+      companionUrl,
+      cloudType: 'nextcloud'
```
Comment (dschmidt) [inline packages/web-app-importer/src/extensions.ts:99]: make this configurable

File: packages/web-pkg/src/apps/types.ts [hunk 3]
```diff
@@ -20,3 +21,70 @@ export interface AppNavigationItem {
...
+  extensions?: Ref<Extension[]>
+  initialize?: () => void
+  ready?: () => void
+  mounted?: () => void
+  // TODO: move this to its own type
+  setup?: (args: { applicationConfig: AppConfigObject }) => ClassicApplicationScript
+}
+
+export const defineWebApplication = (args: {
+  setup: (args: { applicationConfig: AppConfigObject }) => ClassicApplicationScript
```
Comment (dschmidt) [inline packages/web-pkg/src/apps/types.ts:87]: move args type to its own named type

**Review by JammingBen — APPROVED** [review]
🥳

**Comment by dschmidt** [comment]
hoooray 🎉

**Pull request merged**